packages/ibm-products/src/components/TagOverflow/TagOverflowModal.js

```diff
@@ -0,0 +1,134 @@
+// LICENSE file in the root directory of this source tree.
+//
+
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import cx from 'classnames';
+
+import {
+  ComposedModal,
+  ModalHeader,
+  ModalBody,
+  Search,
+  Tag,
+} from '@carbon/react';
+
+import { pkg } from '../../settings';
+import { prepareProps } from '../../global/js/utils/props-helper';
+import { usePortalTarget } from '../../global/js/hooks/usePortalTarget';
+
+const componentName = 'TagOverflowModal';
+const blockClass = `${pkg.prefix}--tag-overflow-modal`;
+
+// Default values for props
+const defaults = {
+  // marked as required by TagSet if needed, default used to satisfy <Search /> component
+  searchLabel: '',
+};
+
+export const TagOverflowModal = ({
+  // The component props, in alphabetical order (for consistency).
+
+  allTags,
+  className,
+  title,
+  onClose,
+  open,
+  portalTarget: portalTargetIn,
+  searchLabel = defaults.searchLabel,
+  searchPlaceholder,
+
+  // Collect any other property values passed in.
+  ...rest
+}) => {
+  const [filteredModalTags, setFilteredModalTags] = useState([]);
+  const [search, setSearch] = useState('');
+  const renderPortalUse = usePortalTarget(portalTargetIn);
+
+  useEffect(() => {
+    let newFilteredModalTags = [];
+    if (open) {
+      if (search === '') {
+        newFilteredModalTags = allTags.slice(0);
+      } else {
+        const lCaseSearch = search.toLocaleLowerCase();
+
+        allTags.forEach((tag) => {
+          const dataSearch = tag['data-search']
+            ?.toLocaleLowerCase()
+            ?.indexOf(lCaseSearch);
+          const labelSearch = tag.label
+            ?.toLocaleLowerCase()
+            ?.indexOf(lCaseSearch);
+
+          if (dataSearch > -1 || labelSearch > -1) {
+            newFilteredModalTags.push(tag);
+          }
+        });
+      }
+    }
+    setFilteredModalTags(newFilteredModalTags);
+  }, [allTags, open, search]);
+
+  const handleSearch = (ev) => {
+    setSearch(ev.target.value || '');
+  };
+
+  return renderPortalUse(
+    <ComposedModal
+      {
+        // Pass through any other property values as HTML attributes.
+        ...rest
+      }
+      containerClassName={`${blockClass}__container`}
+      className={cx(className, `${blockClass}`)}
+      size="sm"
+      {...{ open, onClose }}
+    >
+      <ModalHeader
+        className={`${blockClass}__header`}
+        closeModal={onClose}
+        title={title}
+      >
+        <Search
+          data-modal-primary-focus
+          className={`${blockClass}__search`}
+          labelText={searchLabel}
+          placeholder={searchPlaceholder}
+          onChange={handleSearch}
+          size="lg"
+        />
+      </ModalHeader>
+      <ModalBody className={`${blockClass}__body`} hasForm>
+        {filteredModalTags.map(({ label, ...other }, index) => (
+          <Tag {...other} filter={false} key={`all-tags-${index}`}>
```

Comment:
as a convention it's recommended to not use an index like this as a key

https://react.dev/learn/rendering-lists#why-does-react-need-keys

the tags already have an `id` attribute so i would just continue to use that

Reply:
Replaced index with id attribute.